provider/datagen/src/transform/cldr/source.rs

```diff
@@ -103,12 +99,24 @@ impl<'a> CldrDirLang<'a> {
         for<'de> S: serde::Deserialize<'de> + 'static + Send + Sync,
     {
         self.0
-            .read_and_parse_json(&format!("{}/{}/{}", self.1, lang, file_name))
+            .read_and_parse_json(&format!("{}/{}/{}", self.dir()?, lang, file_name))
     }
 
     pub fn list_langs(&self) -> Result<impl Iterator<Item = LanguageIdentifier>, DataError> {
-        Ok(self.0.list(&self.1)?.into_iter().map(|path| {
-            LanguageIdentifier::from_str(&path.file_name().unwrap().to_string_lossy()).unwrap()
-        }))
+        Ok(self
+            .0
+            .list(&self.dir()?)?
+            .into_iter()
+            .map(|path| LanguageIdentifier::from_str(&path).unwrap()))
+    }
+
+    fn dir(&self) -> Result<String, DataError> {
```

Comment:
Praise: This is a nice clean solution that centralizes the problem more than we had previously